### PR TITLE
feat: add maximum image dimension to be processed by the thumbnailer

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
General hardening: limit input image size when processing thumbnails

## TODO
- [x] test with different image formats - especially jpeg and geobra

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: services/thumbnails/pkg/thumbnail/thumbnail.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package thumbnail
 
 import (
 	"bytes"
+	errors2 "github.com/owncloud/ocis/v2/services/thumbnails/pkg/errors"
```
Comment (dragonchaser) [inline services/thumbnails/pkg/thumbnail/thumbnail.go:5]: errors2 is not so nice, could we use something like `thumbnailerErrors`?

File: services/thumbnails/pkg/thumbnail/thumbnail_test.go [hunk 2]
```diff
@@ -1,8 +1,12 @@
 package thumbnail
 
 import (
+	errors2 "github.com/owncloud/ocis/v2/services/thumbnails/pkg/errors"
```
Comment (dragonchaser) [inline services/thumbnails/pkg/thumbnail/thumbnail_test.go:4]: errors2 is not so nice, could we use something like thumbnailerErrors?

File: services/thumbnails/pkg/thumbnail/thumbnail_test.go [hunk 3]
```diff
@@ -126,10 +132,59 @@ func TestPrepareRequest(t *testing.T) {
 				return
 			}
 
-			// func's are not reflactable, ignore
+			// funcs are not reflactable, ignore
 			if diff := cmp.Diff(tt.want, got, cmpopts.IgnoreFields(Request{}, "Processor")); diff != "" {
 				t.Errorf("PrepareRequest(): %v", diff)
```
Comment (dragonchaser) [inline services/thumbnails/pkg/thumbnail/thumbnail_test.go:137]: can we move this to the errrors pkg aswell?

**Review by dragonchaser — CHANGES_REQUESTED** [review]
while you are at it, could you please move the other errors to the errors package aswell?

File: services/thumbnails/pkg/thumbnail/thumbnail.go (hunk 1 above)
Comment (DeepDiver1975) [reply services/thumbnails/pkg/thumbnail/thumbnail.go:5]: autogenerated by ide :shrug: ...

**Review by dragonchaser — APPROVED** [review]

**Review by dragonchaser — APPROVED** [review]

**Pull request merged**